text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,610 @@
+2. Tabs hooks needs to be moved from React-utils to Pando React
+
+## Drawbacks
+
+### Why should we _not_ do this?
+
+There are tradeoffs to choosing any path. Regarding implentation cost (both in
+term of code size and complexity), there is not much there since we have all the
+components already built out in the Headless-styles sandbox with only some fine-tuning
+that should naturally be made for more robust situations. Likewise, we have had
+a handfull of new people interested in helping contribute to a component library
+which would also speed things up and reduce cost.
+
+Can this library be implemented in our users space? Yes, since most of our teams
+choose to use React, it should only help. However, it should be noted that this
+creates favoritism to React which is typically a negative point since engineering
+should never use tooling from a bias but instead as a result of what is the
+"best tool for the job". It is literally impossible that React will always be the
+best tool for Pluralsight (and it may already be so, we are just choosing not to
+see it?). By nature, one day this specific library will be obsolete.
+
+What would be the impact on teaching people about Pando React? Minimal. The biggest
+hurdle will probably be teaching people about more advanced/preferred React patterns
+based on performance and code scalability. This is only due to the larger amount
+of product code, 3rd party components, and libraries that exist in the world that
+do not use good React practices/anti-patterns (i.e. all in one components, etc.).
+However, I don't think this will really be that big of a deal?
+
+What about integration of this library with other existing features? Everything
+in the [Pando Ecosystem](https://github.com/pluralsight/pando/tree/main/packages/headless-styles#what-role-does-this-package-play-in-pando) should naturally be able
+to co-exists without requiring any additional work from the developer experience.
+This also fulfills our [Commitment to Stability](https://design.pluralsight.com/docs/next/learn/about-pando/versioning#commitment-to-stability).
+
+What is the cost of migrating to Pando React from existing solutions (including Classic)?
+By nature, Pando will be a breaking change since it is fundamentally different down
+to the Web UI Kit (design options). However, it should allow the users to migrate
+easier since it would mostly be updating prop names and values on already existing components.
+Likewise, it should allow the users depenedency list to dramatically decrease since
+they will only need a single package `@pluralsight/react` vs. a package for every single
+component used and it's required dependecies (i.e. Classic, MUI, Chakra, etc.).
+
+Ultimately, the pro's seem to outweigh the cons as long as users understand this
+library is not meant to be the end-game to Pando (Headless-styles is). Headless-styles
```

Comment:
I feel like it might be helpful to change how we talk and think about the difference between `headless-styles` and this component library and the goals of each. I don't think that it should be seen as a temporary solution, or a stepping stone. 

I fully agree that having the core and source of truth be `headless-styles` is the way to go, but right now most teams _do_ use react and as we've all discussed at length, having a react-specific package will be beneficial for a lot of reasons. But since the implementation is framework agnostic, that still allows for growth and change without locking everyone in to react so that as teams grow and technologies change, integrating `headless-styles` with them can become easier. And for common use-cases like this where most teams do use the same tech for now, I think that it's a really good compromise between future flexibility and near-term adoption, and I don't see a need to brand it as the "less ideal" solution.

I think that `headless-styles` could be seen as the core/source of truth, and this component library can be seen as one official "blessed" integration with that source of truth. _"You can write your own if you want or need to, but here's one way that does everything the way we expect it to"_. In C# terms, it's like `headless-styles` is the abstract base class, and this react component library is a derived class. It utilizes all of the building blocks provided by the base class to produce the expected result in React. One could then write another derived class for Vue, svelte, etc. if needed.

Component libraries are popular for a reason and they can enable teams to build fast without worrying about writing tons of plumbing code. It also could allow some implementation details to change without everyone needing to write them manually.

I know of several popular packages that have a framework-agnostic "core" with framework-specific packages utilizing the underlying core package. [Tanstack query](https://tanstack.com/query/latest) is one such example: There is a core package that serves as the source of truth for the core functionality of the package, and framework specific libraries (`react-query`, `vue-query`, `svelte-query`). The framework-specific packages are essentially just plugging the package into the framework, and it enables it to be easy to integrate into a new framework, but that work isn't all put on to the end user. I think that there's a lot of value in this approach; treat `headless-styles` as the core source of implementation for any framework, but with this react package we can see how it can work to have a framework shim without compromising the use of `headless-styles` as the source of truth.

Reply:
@josh-degraw I really appreciate this detailed perspective and just wanted to note that I fully support us changing the branding around the components package role in the ecosystem and want to make sure that people see this statement.

Having said that - I won't update the RFC so this comment/perspective can stay in the thread with the correct context, but will 100% agree to changing the branding to follow the strategy you mentioned which I am in full agreement with.